pkg/arvo/tests/sys/zuse/html.hoon

```diff
@@ -89,4 +90,745 @@
+      ['{a: "b"}' "n_object_unquoted_key"]
+      ['{"a":"a' "n_object_unterminated-value"]
+      ['{ "foo" : "bar", "a" }' "n_object_with_single_string"]
+      ['{"a":"b"}#' "n_object_with_trailing_garbage"]
+      [' ' "n_single_space"]
+      ['["\\uD800\\"]' "n_string_1_surrogate_then_escape"]
+      ['["\\uD800\\u"]' "n_string_1_surrogate_then_escape_u"]
+      ['["\\uD800\\u1"]' "n_string_1_surrogate_then_escape_u1"]
+      ['["\\uD800\\u1x"]' "n_string_1_surrogate_then_escape_u1x"]
+      ['[√©]' "n_string_accentuated_char_no_quotes"]
+      ['["\\x00"]' "n_string_escape_x"]
+      ['["\\\\\\"]' "n_string_escaped_backslash_bad"]
+      ['["\\\09"]' "n_string_escaped_ctrl_char_tab"]
+      ['["\\üåÄ"]' "n_string_escaped_emoji"]
+      ['["\\"]' "n_string_incomplete_escape"]
+      ['["\\u00A"]' "n_string_incomplete_escaped_character"]
+      ['["\\uD834\\uDd"]' "n_string_incomplete_surrogate"]
+      ['["\\uD800\\uD800\\x"]' "n_string_incomplete_surrogate_escape_invalid"]
+      ['["\\uÂ"]' "n_string_invalid-utf-8-in-escape"]
+      ['["\\a"]' "n_string_invalid_backslash_esc"]
+      ['["\\uqqqq"]' "n_string_invalid_unicode_escape"]
+      ['["\\Â"]' "n_string_invalid_utf8_after_escape"]
+      ['[\\u0020"asd"]' "n_string_leading_uescaped_thinspace"]
+      ['[\\n]' "n_string_no_quotes_with_bad_escape"]
+      ['"' "n_string_single_doublequote"]
+      ['[\'single quote\']' "n_string_single_quote"]
+      ['abc' "n_string_single_string_no_double_quotes"]
+      ['["\\' "n_string_start_escape_unclosed"]
+      ['["new' "n_string_unescaped_newline"]
+      ['line"]' "n_string_unescaped_newline"]
+      ['["\09"]' "n_string_unescaped_tab"]
+      ['"\\UA66D"' "n_string_unicode_CapitalU"]
+      ['""x' "n_string_with_trailing_garbage"]
+      ['[‚Å†]' "n_structure_U+2060_word_joined"]
+      ['Ôªø' "n_structure_UTF8_BOM_no_data"]
+      ['<.>' "n_structure_angle_bracket_."]
+      ['[<null>]' "n_structure_angle_bracket_null"]
+      ['[1]x' "n_structure_array_trailing_garbage"]
+      ['[1]]' "n_structure_array_with_extra_array_close"]
+      ['["asd]' "n_structure_array_with_unclosed_string"]
+      ['a√•' "n_structure_ascii-unicode-identifier"]
+      ['[True]' "n_structure_capitalized_True"]
+      ['1]' "n_structure_close_unopened_array"]
+      ['{"x": true,' "n_structure_comma_instead_of_closing_brace"]
+      ['[][]' "n_structure_double_array"]
+      [']' "n_structure_end_array"]
+      ['Ôª{}' "n_structure_incomplete_UTF8_BOM"]
+      ['Â' "n_structure_lone-invalid-utf-8"]
+      ['[' "n_structure_lone-open-bracket"]
+      ['["a\00a"]' "n_string_unescaped_crtl_char"]
+      ['["\\00"]' "n_string_backslash_00"]
+    ==
+:: TODO: de-json is accepting a slew of number formats it shouldn't.
+:: Tracking issue here: https://github.com/urbit/urbit/issues/1775
+:: Re-enable this test by removing the disable- prefix
+++  disable-test-reject-invalid-numbers
+    %-  zing
+    %-  turn
+    :_  |=  [exmpl=@t name=tape]
+          ^-  tang
```

Comment:
Indentation here as well.

An idea: since you have several tests that involve this pattern of traversing a list of test specs, you could factor out the common elements.

A JSON test-specification type:

```
+$  json-tests-spec  [data=cord name=tape]
```

A specification arm:

```
++  invalid-numbers-spec
  ^-  (list json-test-spec)
  :~  ['123\00' "n_multidigit_number_then_00"]
        ...
  ==
```

A single definition of the spec-runner gate, and then a bunch of test arms that just call the gate on the appropriate input. That might be a bit too factored, you could define the type and gate separately, and then define the test arms as calls to the gate on literal data structures. In any case, a little less duplication would be nice here.